### PR TITLE
[FP-18] Login Functionality with Redux

### PR DESCRIPTION
**Pull request by thisisofficial**
### Description

This feature was a long time comming, and it's finally done.
With this, the backend for our auth system gets a request from our ap to return
encrypted credentials for us to use.

- Added a mutation query to the backend API
- Configured our app-wide store to use new fetch request

### Testing Checklist

- [x] Test cases have been added/updated
- [x] Tested locally in browser (at least one other than Chrome)
- [x] All tests pass

### What JIRA Ticket(s) does this PR address?

- [FP-18](https://cd95.atlassian.net/browse/FP-18)

### CODE REVIEW

**Review by caarlosdamian — CHANGES_REQUESTED** [review]
Check test are failing

File: src/pages/Login/Login.tsx [hunk 1]
```diff
@@ -1,17 +1,35 @@
 /* eslint-disable jsx-a11y/label-has-associated-control */
+import { useNavigate } from 'react-router-dom';
 import { useForm, SubmitHandler } from 'react-hook-form';
+import { useDispatch } from 'react-redux';
+import { FetchBaseQueryError } from '@reduxjs/toolkit/dist/query';
+import { SerializedError } from '@reduxjs/toolkit';
 import img from '../../assets/image_login.png';
 import './Login.scss';
```
Comment (caarlosdamian) [inline src/pages/Login/Login.tsx]: move this to line 11

File: src/routes/routes.tsx [hunk 2]
```diff
@@ -19,7 +19,10 @@ export const Router = () => {
           <Route
             path={item.url}
             element={
-              <ProtectedRoute user={loginState.value} redirectPath="/login">
+              <ProtectedRoute
+                user={loginState.user.isLogged}
```
Comment (caarlosdamian) [inline src/routes/routes.tsx]: use just isLogged

File: src/routes/routes.tsx [hunk 3]
```diff
@@ -8,7 +8,7 @@ import { items } from '.';
 
 export const Router = () => {
   interface IState {
-    loginSlice: { value: boolean };
+    loginSlice: { user: { token: string; isLogged: boolean } };
   }
 
   const loginState = useSelector((state: IState) => state.loginSlice);
```
Comment (caarlosdamian) [inline src/routes/routes.tsx]: use destructuring const {isLogged} = useSelector((state: IState) => state.loginSlice.user);

File: src/pages/Login/Login.test.tsx [hunk 4]
```diff
@@ -1,10 +1,21 @@
...
 import user from '@testing-library/user-event';
+import { Provider } from 'react-redux';
+import { BrowserRouter, Route, Routes } from 'react-router-dom';
+import { store } from '../../redux/Store';
 import { Login } from './Login';
 
 describe('Login  page component render correctly', () => {
   beforeEach(() => {
-    render(<Login />);
+    render(
```
Comment (caarlosdamian) [inline src/pages/Login/Login.test.tsx]: this is already in test-utils.tsx please remove it

File: src/pages/Login/Login.tsx (hunk 1 above)
Comment (caarlosdamian) [reply src/pages/Login/Login.tsx]: this already isnt good move it to line styling imports should be last

File: src/routes/routes.tsx (hunk 3 above)
Comment (caarlosdamian) [reply src/routes/routes.tsx]: this still the same use destructuring

File: src/routes/routes.tsx (hunk 2 above)
Comment (caarlosdamian) [reply src/routes/routes.tsx]: please check comments

**Review by caarlosdamian — CHANGES_REQUESTED** [review]

File: src/pages/Login/Login.test.tsx (hunk 4 above)
Comment (thisisofficial) [reply src/pages/Login/Login.test.tsx]: > this is already in test-utils.tsx please remove it

I cant find said file. Care to direct me to it?.
It was also changed to pass tests, so now its a bit different.

**Review by caarlosdamian — APPROVED** [review]

**Pull request merged**